"""
    Задача решается при помощи рекурсии.
    Рекурсия - явление, при котором функция вызывает саму себя.

    Проблемные места задания:
    1. Глубина рекурсии (Ошибка: RecursionError: maximum recursion depth exceeded)
        Решение:
                import sys
                sys.setrecursionlimit(желаемая_глубина)
    2. Время выполнения функции
        Решение:
                Анализировать дебагом функцию и смотреть, что она делает и как, пытаться понять суть.
"""